tidy3d/components/medium.py

```diff
@@ -1267,6 +1336,257 @@ def elements(self) -> Dict[str, IsotropicMediumType]:
+        if np.any(np.linalg.eigvals(cond_sym) < -fp_eps):
+            raise ValidationError("Main diagonal of provided conductivity tensor is not >= 0.")
+
+        return val
+
+    @classmethod
+    def from_diagonal(cls, xx: Medium, yy: Medium, zz: Medium, rotation: RotationType):
+        """Construct a fully anisotropic medium by rotating a diagonally ansisotropic medium.
+
+        Parameters
+        ----------
+        xx : Medium
+            Medium describing the xx-component of the diagonal permittivity tensor.
+        yy : Medium
+            Medium describing the yy-component of the diagonal permittivity tensor.
+        zz : Medium
+            Medium describing the zz-component of the diagonal permittivity tensor.
+        rotation : RotationType
+                Rotation applied to diagonal permittivity tensor.
+
+        Returns
+        -------
+        :class:`FullyAnisotropicMedium`
+            Resulting fully anisotropic medium.
+        """
+
+        permittivity_diag = [
+            [xx.permittivity, 0, 0],
+            [0, yy.permittivity, 0],
+            [0, 0, zz.permittivity],
+        ]
+
+        conductivity_diag = [
+            [xx.conductivity, 0, 0],
+            [0, yy.conductivity, 0],
+            [0, 0, zz.conductivity],
+        ]
+
+        permittivity = rotation.rotate_tensor(permittivity_diag)
+        conductivity = rotation.rotate_tensor(conductivity_diag)
+
+        return cls(permittivity=permittivity, conductivity=conductivity)
+
+    @cached_property
+    def eps_sigma_diag(self) -> Tuple[Tuple[float, float, float], Tuple[float, float, float]]:
+        """Main components of permittivity and conductivity tensors."""
+
+        perm_diag, vecs = np.linalg.eig(self.permittivity)
+        cond_diag = np.diag(np.matmul(np.transpose(vecs), np.matmul(self.conductivity, vecs)))
+
+        return (perm_diag, cond_diag)
+
+    @ensure_freq_in_range
+    def eps_model(self, frequency: float) -> complex:
+        """Complex-valued permittivity as a function of frequency."""
+        perm_diag, cond_diag = self.eps_sigma_diag
+
+        eps_diag = AbstractMedium.eps_sigma_to_eps_complex(perm_diag, cond_diag, frequency)
+        return np.mean(eps_diag)
+
+    @ensure_freq_in_range
+    def eps_diagonal(self, frequency: float) -> Tuple[complex, complex, complex]:
+        """Main diagonal of the complex-valued permittivity tensor as a function of frequency."""
+
+        perm_diag, cond_diag = self.eps_sigma_diag
+
+        if np.isscalar(frequency):
+            eps_diag = AbstractMedium.eps_sigma_to_eps_complex(perm_diag, cond_diag, frequency)
+        else:
+            eps_diag = AbstractMedium.eps_sigma_to_eps_complex(
+                perm_diag[:, None], cond_diag[:, None], frequency
+            )
+
+        return eps_diag
+
+    def eps_comp(self, row: Axis, col: Axis, frequency: float) -> complex:
+        """Single component the complex-valued permittivity tensor as a function of frequency."""
```

Comment:
copy in the docstring w. parameters and returns

Reply:
should I do the same for `eps_comp` in old `AnisotropicMedium`?

Reply:
Yea, basically if it's going in the docs (public method) we should include the parameters / returns

Reply:
fixed